### ISSUE TITLE
Migration to Java 17 (LTS)

### ISSUE THREAD

**Issue by Doomsdayrs** [ATLauncher/ATLauncher#601]
**Is your feature request related to a problem? Please describe.**
As of Minecraft 1.17, Java 16 was used.
As of Minecraft 1.18, Java 17 was used.
Minecraft has moved on from the older versions of java, ATLauncher should follow suite with moving to Java 17 (LTS).

**Additional context**
- AWT and Swing are both separate libraries from Java now, so they will have to be bundled as so
- Significant testing will have to occur to ensure that the various older instances work as expected.

**Comment by RyanTheAllmighty** [comment]
Yeah I don't think this is a good idea honestly. ~90-95% of the userbase is still using Java 8 and having the launcher just flat out stop for them on an update is not something we can really do, unless you have a plan to address that?

**Comment by Doomsdayrs** [comment]
> Yeah I don't think this is a good idea honestly. ~90-95% of the userbase is still using Java 8 and having the launcher just flat out stop for them on an update is not something we can really do, unless you have a plan to address that?

I want it running with 0 changes to user experience. So I will be testing with *every* single feature and ensuring it runs on all devices.

In the PR, I am beginning my tests to ensure stability and user experience is consistent with the current launcher. And noting any and all bugs for me to fix myself.

I will also be testing how users will handle such an update, especially if they are coming from java 8.

**Comment by RyanTheAllmighty** [comment]
Trying to somehow tell users and guide them to upgrade their Java version to 17 on their PC is not sounding like fun trying to tell people and support people in doing so.

This honestly to me sounds like a massive hassle and support burden for very tiny benefit

**Comment by Doomsdayrs** [comment]
> Trying to somehow tell users and guide them to upgrade their Java version to 17 on their PC is not sounding like fun trying to tell people and support people in doing so.

That is why I intend to test how the experience goes. I do not intend to push a change that would ruin user experience completely. 

Minecraft has already done this migration with 0 user backlash. I have yet to see how this migration can not be done.

> This honestly to me sounds like a massive hassle and support burden for very tiny benefit

The hassle is all on me, the support burden is subjective as it stands with a lack of testing.

**Comment by RyanTheAllmighty** [comment]
> Minecraft has already done this migration with 0 user backlash.

Yes but one thing to keep in mind is Minecraft launcher doesn't run on Java, and they bundle the JRE with most Minecraft versions anyway, so there would be no issue.

Since we require Java already to even run the launcher, trying to get those 90-95% of users on Java 8 to upgrade is my concern. Just don't want you wasting your time trying to find issues with the launcher with the most vital piece and blocker not being resolved.

I'm happy to do the change if it all works out, but the core issue needs to be solved super well else it'll honestly not be worth the hassle to do in my eyes

**Comment by Doomsdayrs** [comment]
Ah, Now I understand what you mean. I suppose I will close my PR for now and concern myself elsewhere with my passion.

The clock is ticking in the background then as Java 8 Extended support ends 2030.

For the time being, Would changing the JRE from java 8 to 17 in https://github.com/ATLauncher/ATLauncher/blob/master/packaging/windows-setup/installer.iss be possible?

**Comment by RyanTheAllmighty** [comment]
If this launcher is still around in 2030 I'll be shocked.

I did start the look at upgrading that in the installer, but never actually did it. Probably somewhere to start maybe

**Comment by Doomsdayrs** [comment]
> If this launcher is still around in 2030 I'll be shocked.

You undervalue your launcher far too much.

> I did start the look at upgrading that in the installer, but never actually did it. Probably somewhere to start maybe

I agree, I already have the flatpak running under java 11 at least. So it should be time to get everyone on the LTS in the background, provide news in regards to it, then wait 1 year before releasing the java 17 upgrade.

**Comment by Doomsdayrs** [comment]
Or if you want to be more gradual, A change to java 11 first would work as well. At least bump so we have the security and improvements.

**Comment by RyanTheAllmighty** [comment]
Nah straight to 17 should be fine. I do all my testing on Java 17 anyway, so it should be okay.

The installer is a completely separate process to the launcher updates, so if people have issues, can just revert the installer back

**Comment by Doomsdayrs** [comment]
Just a note in regards to Applet, It is marked for removal and will not exist soon.

Which means the legacy launcher workaround might not work properly.

**Comment by Doomsdayrs** [comment]
@RyanTheAllmighty Where do we stand now?

**Comment by RyanTheAllmighty** [comment]
Ah I still need to get the prompt added back in. Leave it with me, I'll do that tonight, I want to do a release shortly anyway
